File: rerun_cpp/src/rerun/components/material_ext.cpp
#include "material.hpp"

// Uncomment for better auto-complete while editing the extension.
// #define EDIT_EXTENSION

namespace rerun {
    namespace components {

#ifdef EDIT_EXTENSION
        struct MaterialExt {
#define Material MaterialExt

            // <CODEGEN_COPY_TO_HEADER>

            static Material from_albedo_factor(rerun::datatypes::Rgba32 color) {
                return Material(color);
            }

            // </CODEGEN_COPY_TO_HEADER>
        };
#endif
    } // namespace components
} // namespace rerun

File: crates/store/re_types/src/components/material_ext.rs
use crate::datatypes::Rgba32;

use super::Material;

impl Material {
    /// A new material using a given color multiplier.
    #[inline]
    pub fn from_albedo_factor(color: impl Into<Rgba32>) -> Self {
        Self(crate::datatypes::Material::from_albedo_factor(color))
    }
}
